service/common/src/main/java/org/apache/polaris/service/catalog/iceberg/IcebergCatalog.java

```diff
@@ -798,13 +790,12 @@ private Page<Namespace> listNamespaces(Namespace namespace, PageToken pageToken)
                 PolarisEntityType.NAMESPACE,
                 PolarisEntitySubType.NULL_SUBTYPE,
                 pageToken);
-    List<PolarisEntity.NameAndId> entities =
-        PolarisEntity.toNameAndIdList(listResult.getEntities());
-    List<Namespace> namespaces = PolarisCatalogHelpers.nameAndIdToNamespaces(catalogPath, entities);
     return listResult
-        .getPageToken()
-        .map(token -> new Page<>(token, namespaces))
-        .orElseGet(() -> Page.fromItems(namespaces));
+        .getEntities()
+        .map(
+            record ->
```

Comment:
Why `record`?

Reply:
`record` comes from the type name of `EntityNameLookupRecord`

Reply:
This change is not about simplification, but about controlling the propagation of the page token inside the `Page` class.